Light visualiser Base v2

danieldresser opened the pull request:
This should now have addressed all of John's comments on the previous version of this pull request ( there is enough change that it seemed like it's simpler to make a new pull request ).

There probably are still some updates required to the Arnold and Appleseed stuff - John, we should chat about whether you would be able to take this the rest of the way to getting merged.

est77 commented:
I can do the appleseed part if you want.

About area lights, it's already possible to have them in appleseed by assigning an OSL shader with an emission() closure to an object. It works for now because we don't have procedurals and the order in which light sources are output is not really important.

johnhaddon commented:
Looks like we have some failing tests - could you investigate Daniel?

danieldresser commented:
Did some investigation of failing tests.  Started by getting sidetracked by testWedge, which was disabled on travis and failing locally.  Adding a bound to the ImageStats appears to have gotten it working both locally and on travis.  This means I've now got GafferRenderManTest completely passing locally.  The failure on travis appears to be because I'm depending on the support for renderer prefixes you added in Cortex-9.1.0.  We probably can't just hack around this one, and will need to update the Cortex version we test against?

The other failure is in GafferSceneTest.ParametersTest where it's trying to set parameters on a light object that is no longer stored this way.  Do you think we need to try and set up some short term support for lights on the Parameters node, or do we wait for general shader support for parameters, and just delete this test for the moment?

johnhaddon commented:
Thanks for looking into that, and for making the bonus fix in testWedge. I'll update gafferDependencies with the latest Cortex and kick off a build for that tonight - won't be able to upload it till the morning though. I think when we discussed the Parameters node previously, the view was that in practice we're not using the it in anger with lights at present, so let's remove that test. We will need to note the backwards compatibility in the release notes though, so if you could put a handy reminder in the commit message that'd be good.

danieldresser commented:
OK, I've now added a commit that removes the light test from GafferScene.ParametersTest, and added a note to the light node commit message.  Hopefully the tests will all go through with the new Cortex version.

johnhaddon commented:
Closing this since #1590 contains the latest-and-greatest revision of this stuff.
